### PR TITLE
make sure we don't introduce backward incompatibility changes in the same minor series

### LINKED ISSUE #1338: e2e cases to verify upgrading compatiabilty
## Feature Request

ref: https://github.com/pingcap/tidb-operator/issues/1335

**Is your feature request related to a problem? Please describe:**

We should guarantee the compatibility when upgrade `tidb-operator` and different branch/release may require different level of compatibility, e.g. in release branch, we want to make sure that upgrade tidb-operator will never trigger rolling-update or error of the TiDB cluster managed by it. Therefore, we need some test cases to cover this.

**Describe the feature you'd like:**

Add e2e cases to verify upgrading compatibility.

**Describe alternatives you've considered:**

Add new case in stability test to verify. This is not ideal because stability test is not tend to run for each commit.

@cofyc PTAL

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes #1338
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/serial.go [hunk 1]
```diff
@@ -111,109 +110,6 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 		}
 	})
 
-	ginkgo.Context("tidb-operator with default values", func() {
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go:114]: moved to stability.go

File: tests/e2e/tidbcluster/serial.go [hunk 2]
```diff
@@ -1136,7 +1032,63 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 				klog.Infof("confirm tikv pods haven't been changed this time")
 				return false, nil
 			})
-			framework.ExpectError(err, "expect tikv and pd haven't been changed for 5 minutes")
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect tikv and pd haven't been changed for 5 minutes")
+		})
+	})
+
+	ginkgo.Context("upgrading tidb-operator in the same minor series should not trigger rolling-update", func() {
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go:1039]: this case can be covered by unit or integration tests but that requires us to maintain some test data.  it's easier to have a simple e2e test for now.

**Comment by Yisaer** [comment]
@cofyc  do we need cherry-pick to 1.1

**Comment by cofyc** [comment]
yes

**Pull request merged**